docs-src/java/how-to-use-a-custom-payload-codec-in-java.md

```diff
@@ -0,0 +1,61 @@
+---
+id: how-to-use-a-custom-payload-codec-in-java
+title: How to use a custom Payload Codec in Java
+sidebar_label: Custom Payload Codec
+description: Create a custom implementation of the `PayloadCodec` and use it in a `CodecDataConverter` to set a custom Data Converter.
+tags:
+  - Java
+  - developer-guide
+---
+
+**Create a custom Payload Codec**
+
+Create a custom implementation of the [`PayloadCodec`](https://www.javadoc.io/static/io.temporal/temporal-sdk/1.18.1/io/temporal/payload/codec/PayloadCodec.html) and use it in a [`CodecDataConverter`](https://www.javadoc.io/doc/io.temporal/temporal-sdk/latest/io/temporal/common/converter/CodecDataConverter.html)) to set a custom Data Converter.
+
+The Payload Codec does byte to byte conversion, and must be set with a Data Converter to do the conversion to bytes and values.
+
+Define custom encryption/compression logic in your `encode` method, and decryption/decompression logic in your `decode` method.
+
+```java
+public class YourCustomPayloadCodec implements PayloadCodec {
+    @NotNull
+    @Override
+    public List<Payload> encode(@NotNull List<Payload> payloads) {
+        //your encryption/compression logic
+    }
+
+    @NotNull
+    @Override
+    public List<Payload> decode(@NotNull List<Payload> payloads) {
+        //your decryption/decompression logic
+    }
+    //...
+}
+```
+
+You can also create a remote HTTP server (called Codec Server) to run the encryption and decryption through the custom `PayloadCodec`, and expose endpoints that you can use with WebUI and tctl to see decrypted data.
```

Comment:
```suggestion
You can also create a remote HTTP server (called a codec server) to run the encryption and decryption through the custom `PayloadCodec`, and expose endpoints that you can use with Web UI and tctl to see decrypted data.
```
But see Dmitry's comment and apply any change both here and near the end of `docs-src/go/how-to-use-a-custom-payload-codec-in-go.md`.